pep-0517.txt

```diff
@@ -166,48 +166,71 @@ The build backend object is expected to have attributes which provide
 some or all of the following hooks. The common ``config_settings``
 argument is described after the individual hooks::
 
-  def get_build_requires(config_settings):
+  def build_sdist(source_directory, sdist_directory, config_settings):
+      ...
+
+MUST build an unpacked sdist from the ``source_directory`` and place it into the
+specified ``sdist_directory``. If the ``source_directory`` is already an
+unpacked sdist (as opposed to a VCS source tree) then it SHOULD simply copy the
```

Comment:
We don't specify how to tell there are only three states that a thing can be in, it can be a wheel, a sdist, or something that can be turned into a wheel or sdist. While I used the word "VCS source tree" it doesn't literally have to be a VCS tree. It could just be a VCS-less directory on disk, that's fine. What requirements any particular build tool might place on what it requires of a target project, is really up to that particular build tool. Some might work just fine with a sdist generation step that is nothing more than ``shutil.copytree`` in which case they don't really need to differentate between a "VCS source tree" and a "sdist". Another project might, when in a VCS tree, dynamically compute a version number based on the git tags and "Bake" that into the sdist (using some internal to them scheme), that project would differentiate and would likely do it by looking at that internal scheme.

This PEP declares what a sdist is, it's the same thing we've had for like 2 decades, just ``s/setup.py/pyproject.toml/``. That's perfectly reasonable for this PEP and I don't think we need to get bogged down in the weeds declaring a sdist 2.0 or anything like that.

As far as when would pip itself call this, as written in this PR it would call that ``build_sdist`` hook anytime it is building from source unless that source is literally a ``foobar-1.0.tar.gz``. This means it'll do it from a git clone or if you unpack a sdist yourself (this is why the hook needs to handle the case of being run in something that was already a sdist) because we can't differentiate between an unpacked sdist and a random git clone (or whatever) without either getting bogged down in a sdist 2.0 or knowing more about what the internals of the build tool.

As far as failure cases go, if the hook fails the build fails.